compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3995,6 +3995,10 @@ private ParserRuleContext getNextRuleForKeywords(ParserRuleContext currentCtx, i
             case OUTER_KEYWORD:
                 return ParserRuleContext.JOIN_KEYWORD;
             case MAP_KEYWORD:
+                parentCtx = getParentContext();
```

Comment:
Need to add `MAP_KEYWORD` as an alternative path for the `TABLE_CONSTRUCTOR_OR_QUERY_START` context.

https://github.com/ballerina-platform/ballerina-lang/blob/94b6e7369f1d810df0189d41572fb8e142fc8e4d/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java#L426-L427